src/vellum/workflows/nodes/displayable/tool_calling_node/tests/test_node.py

```diff
@@ -140,3 +142,75 @@ def test_deployment_definition_release_tag_defaults_to_latest():
 
     # THEN the release_tag should default to "LATEST"
     assert deployment_config.release_tag == "LATEST"
+
+
+def test_tool_calling_node_with_user_provided_chat_history_block(vellum_adhoc_prompt_client):
+    """
+    Test that ToolCallingNode with user-provided chat history block merges user and node messages.
+    """
+
+    # GIVEN a ToolCallingNode with a user-provided chat history block
+    user_chat_history_block = VariablePromptBlock(
+        block_type="VARIABLE",
+        input_variable="chat_history",
+        state=None,
+        cache_config=None,
+    )
+
+    class TestToolCallingNode(ToolCallingNode):
+        ml_model = "gpt-4o-mini"
+        blocks = [user_chat_history_block]
+        functions = [first_function]
+        prompt_inputs = {"chat_history": [ChatMessage(role="USER", text="Hello from user")]}
+        max_prompt_iterations = 1
+
+    def generate_prompt_events(*args: Any, **kwargs: Any) -> Iterator[Any]:
+        from uuid import uuid4
+
+        from vellum.client.types.fulfilled_execute_prompt_event import FulfilledExecutePromptEvent
+        from vellum.client.types.initiated_execute_prompt_event import InitiatedExecutePromptEvent
+        from vellum.client.types.string_vellum_value import StringVellumValue
+
+        execution_id = str(uuid4())
+        events = [
+            InitiatedExecutePromptEvent(execution_id=execution_id),
+            FulfilledExecutePromptEvent(
+                execution_id=execution_id,
+                outputs=[StringVellumValue(value="Hello! I can help you.")],
+            ),
+        ]
+        yield from events
+
+    vellum_adhoc_prompt_client.adhoc_execute_prompt_stream.side_effect = generate_prompt_events
+
+    # AND a state with user-provided chat history
+    class TestState(BaseState):
+        chat_history: List[ChatMessage] = []
+        prompt_iterations: int = 0
```

Comment:
remove this state - the parent workflow does _not_ need to presume initial state